test/src/main/java/org/apache/accumulo/test/functional/ThriftMaxFrameSizeIT.java

```diff
@@ -56,50 +62,95 @@ class TestDefault extends TestMaxFrameSize {
     TestDefault() {
       serverType = ThriftServerType.getDefault();
     }
+
+    @Test
+    public void testDefaultServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestThreadedSelector extends TestMaxFrameSize {
     TestThreadedSelector() {
       serverType = ThriftServerType.THREADED_SELECTOR;
     }
+
+    @Test
+    public void testThreadedSelectorServerFrameSize() throws Exception {
```

Comment:
Making these separate tests in each separate `@Nested` class starts defeating the reason these are nested. Plus, all of them except the Ssl one (which I don't understand why it works) has the same implementation. So it can just stay in the base class. The Ssl one could just be made to use a different base class that doesn't have the same method.

Reply:
I pushed another commit that refactors this test further to address this and other issues.